src/main/java/com/hazelcast/jdbc/JdbcConnection.java

```
@@ -0,0 +1,352 @@
+        checkClosed();
+        throw JdbcUtils.unsupported("CallableStatement not supported");
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String sql, int autoGeneratedKeys) throws SQLException {
+        checkClosed();
+        throw unsupportedAutoGeneratedKeys();
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String sql, int[] columnIndexes) throws SQLException {
+        checkClosed();
+        throw unsupportedAutoGeneratedKeys();
+    }
+
+    @Override
+    public PreparedStatement prepareStatement(String sql, String[] columnNames) throws SQLException {
+        checkClosed();
+        throw unsupportedAutoGeneratedKeys();
+    }
+
+    @Override
+    public Clob createClob() throws SQLException {
+        checkClosed();
+        return null;
+    }
+
+    @Override
+    public Blob createBlob() throws SQLException {
+        checkClosed();
+        return null;
+    }
+
+    @Override
+    public NClob createNClob() throws SQLException {
+        checkClosed();
+        return null;
+    }
+
+    @Override
+    public SQLXML createSQLXML() throws SQLException {
+        checkClosed();
+        return null;
+    }
+
+    @Override
+    public boolean isValid(int timeout) throws SQLException {
+        checkClosed();
+        return !isClosed();
+    }
+
+    @Override
+    public void setClientInfo(String name, String value) throws SQLClientInfoException {
+    }
+
+    @Override
+    public void setClientInfo(Properties properties) throws SQLClientInfoException {
+    }
+
+    @Override
+    public String getClientInfo(String name) throws SQLException {
+        checkClosed();
+        return null;
+    }
+
+    @Override
+    public Properties getClientInfo() throws SQLException {
+        checkClosed();
+        return null;
+    }
+
+    @Override
+    public Array createArrayOf(String typeName, Object[] elements) throws SQLException {
```

Since we do not support `createArrayOf` and `createStruct`, it is better to throw the "unsupported" exception, rather than exposing the user to possible NPE .